opencensus/exporters/trace/zipkin/zipkin_exporter.cc

```diff
@@ -0,0 +1,342 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "zipkin_exporter.h"
+
+#include <curl/curl.h>
+#include <rapidjson/prettywriter.h>
+#include <rapidjson/stringbuffer.h>
+#include "absl/strings/str_cat.h"
+
+namespace opencensus {
+namespace exporters {
+namespace trace {
+
+namespace {
+
+constexpr const char *kZipkinLibname = "zipkin";
+constexpr const char *kZkipkinVersion = "2.0";
+
+void SerializeJson(rapidjson::Writer<rapidjson::StringBuffer> *writer,
+                   const ::opencensus::trace::exporter::SpanData &span) {
+  writer->StartObject();
+
+  writer->Key("name");
```

Comment:
at the very least we should (if v2 format) always write span.localEndpoint
if v1 format, span.localEndpoint is copied to every annotation and binary annotation

Either way, without this indexing on service or span name won't work.

Reply:
I updated it to record the localEndpoint now.